src/components/AnimatedCollapsible/index.tsx

```diff
@@ -0,0 +1,152 @@
+import React, {useEffect, useRef} from 'react';
+import type {ReactNode} from 'react';
+import {View} from 'react-native';
+import type {StyleProp, ViewStyle} from 'react-native';
+import Animated, {useAnimatedStyle, useDerivedValue, useSharedValue, withTiming} from 'react-native-reanimated';
+import Icon from '@components/Icon';
+import * as Expensicons from '@components/Icon/Expensicons';
+import {easing} from '@components/Modal/ReanimatedModal/utils';
+import {PressableWithFeedback} from '@components/Pressable';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import CONST from '@src/CONST';
+
+type AnimatedCollapsibleProps = {
+    /** Whether the component is expanded */
+    isExpanded: boolean;
+
+    /** Element that is inside the collapsible area */
+    children: ReactNode;
+
+    /** Header content to display above the collapsible content */
+    header: ReactNode;
+
+    /** Duration of expansion animation */
+    duration?: number;
+
+    /** Additional external style for the container */
+    style?: StyleProp<ViewStyle>;
+
+    /** Style for the header container */
+    headerStyle?: StyleProp<ViewStyle>;
+
+    /** Style for the content container */
+    contentStyle?: StyleProp<ViewStyle>;
+
+    /** Style for the toggle button */
+    expandButtonStyle?: StyleProp<ViewStyle>;
+
+    /** Whether the toggle button is disabled */
+    disabled?: boolean;
+
+    /** Callback for when the toggle button is pressed */
+    onPress: () => void;
+};
+
+function AnimatedCollapsible({isExpanded, children, header, duration = 300, style, headerStyle, contentStyle, expandButtonStyle, onPress, disabled = false}: AnimatedCollapsibleProps) {
+    const theme = useTheme();
+    const styles = useThemeStyles();
+    const height = useSharedValue(0);
+    const isAnimating = useSharedValue(false);
+    const hasExpanded = useSharedValue(false);
+    const isExpandedFirstTime = useRef(false);
+
+    useEffect(() => {
+        if (!isExpanded && !isExpandedFirstTime.current) {
+            return;
+        }
+        if (isExpandedFirstTime.current) {
+            hasExpanded.set(true);
+        } else {
+            isExpandedFirstTime.current = true;
+        }
+    }, [hasExpanded, isExpanded]);
+
+    // Animation for content height and opacity
+    const derivedHeight = useDerivedValue(() => {
```

Comment:
Trying to understand whether this will create new animation instance on every render? and do we need any clean up for interrupted animations?